File: LeetCode Solutions/Problems/845. Longest Mountain in Array.py
class Solution(object):
    def longestMountain(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        left, right = 0, 1
        length = 0
        while(right < len(arr)):
            if(arr[left] >= arr[right]):
                left += 1
                right += 1
            else:
                haveRight = 0
                while(right < (len(arr)-1) and arr[right] < arr[right+1]):
                    right += 1
                while(right < (len(arr)-1) and arr[right] > arr[right+1]):
                    right += 1
                    haveRight = 1
                if(haveRight):
                    length = max(length,right-left+1)
                left = right
                right = left + 1
        return length

File: LeetCode Solutions/Problems/845. Longest Mountain in Array.java
class Solution {
    public int longestMountain(int[] arr) {
        int left = 0;
        int right = 1;
        int length = 0;
        while(right < arr.length){
            if(arr[left] >= arr[right]){
                left += 1;
                right += 1;
            } else{
                int haveRight = 0;
                while(right < (arr.length-1) && arr[right] < arr[right+1]){
                    right += 1;
                }
                while(right < (arr.length-1) && arr[right] > arr[right+1]){
                    right += 1;
                    haveRight = 1;
                }
                if(haveRight != 0){
                    length = Math.max(length,right-left+1);
                }
                left = right;
                right = left + 1;
            }
        }
        return length;
    }
}